Sources/RemoteMessaging/Model/MatchingAttributesPrototypes/StringRangeMatching.swift

```diff
@@ -42,7 +42,12 @@ extension StringRangeMatching {
         let value = jsonMatchingAttribute[RuleAttributes.value] as? String ?? MatchingAttributeDefaults.stringDefaultValue
         let fallback = jsonMatchingAttribute[RuleAttributes.fallback] as? Bool
 
-        self.init(min: min, max: max, value: value, fallback: fallback)
+        self.init(
+            min: min.trimmingBuildNumber,
+            max: max.trimmingBuildNumber,
+            value: value.trimmingBuildNumber,
```

Comment:
These lines exist to avoid breaking existing messages that already define a build number.

This approach means that build number trimming will happen for all `StringRangeMatching` attributes, which is currently `AppVersionMatchingAttribute` and `OSMatchingAttribute`. But, if you would prefer to keep this change limited to the app version attribute, I can override its initializer and push that logic down there instead.

Alternatively, it could be moved into `RangeStringNumericMatchingAttribute` instead. We have a few options, let me know what you prefer. Thanks!

Reply:
The logic we're applying here is specific to the app version attribute, right? I'd rather see it in the `AppVersionMatchingAttribute` to limit its reach. Unless you foresee this adjustment to be useful for other strings?